### PR TITLE
Update robolectric, use Runner class instead of annotations

### PR DESCRIPTION
**Pull request by dschuermann**
* updates robolectric to 3.2.2
* uses Runner class to configure Manifest path and SDK instead of repeating the same annotation again and again

### CODE REVIEW

File: k9mail-library/src/test/java/com/fsck/k9/mail/K9LibRobolectricTestRunner.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+public class K9LibRobolectricTestRunner extends RobolectricTestRunner {
+
+    public K9LibRobolectricTestRunner(Class<?> testClass) throws InitializationError {
+        super(testClass);
+    }
+
+    @Override
+    protected Config buildGlobalConfig() {
+        return new Config.Builder()
+                .setSdk(21)
```
Comment (philipwhiuk) [inline k9mail-library/src/test/java/com/fsck/k9/mail/K9LibRobolectricTestRunner.java]: I don't know much about Robolectric testing really. Do you think this should:

a) Generally match the targetSdkVersion?
b) Generally match the minimumVersion?
c) Just be consistent?

Currently it's 21 and we're building for 22.

I assume there's scenarios where you want to test a particular function and so you need a certain version

File: k9mail-library/src/test/java/com/fsck/k9/mail/K9LibRobolectricTestRunner.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+                .setManifest(Config.NONE)
```
Comment (philipwhiuk) [inline k9mail-library/src/test/java/com/fsck/k9/mail/K9LibRobolectricTestRunner.java]: I think some tests might need a config - it's how you get resources. Again I'm too novice to know whether there's a downside to always providing one.

Comment (dschuermann) [inline k9mail-library/src/test/java/com/fsck/k9/mail/K9LibRobolectricTestRunner.java]: Yup, this is the runner for k9mail-library, there is a different runner for the app which defines a Manifest path here by default. Besides, you can always override this default config via annotations again.

File: k9mail-library/src/test/java/com/fsck/k9/mail/K9LibRobolectricTestRunner.java (hunk 1 above)
Comment (Valodim) [inline k9mail-library/src/test/java/com/fsck/k9/mail/K9LibRobolectricTestRunner.java]: Robolectric only has a few sdk numbers that work here, and the differences are relatively minor. I think it's generally c) here with a dash of a)

Comment (dschuermann) [inline k9mail-library/src/test/java/com/fsck/k9/mail/K9LibRobolectricTestRunner.java]: I would use the ``targetSdkVersion`` which is 22. This is also the default of robolectric if we do not configure anything.

So why even configure a sdk version here? In the past robolectric did not provide all the sdk versions, they were lagging behind Android releases. I would keep the explicit configuration to make it less depended on changes in the build.gradle.

I added a commit to set sdk to 22 in both runners.

You can explicitly override the runner config by adding an annotation again, but before each test had the annotation to use sdk 21, which violates DRY.

**Comment by cketti** [comment]
Awesome. Thanks :+1:

**Pull request merged**